### PR TITLE
chore: update sync repo settings

### PR DESCRIPTION
**Pull request by anguillanneuf**
- turn on Kokoro integration testing as a required check
- update permission rules
- remove java7 branch rules

### CODE REVIEW

File: .github/sync-repo-settings.yaml [hunk 1]
```diff
@@ -15,24 +15,13 @@ branchProtectionRules:
       - units (11)
       - cla/google
       - OwlBot Post Processor
-  - pattern: java7
-    isAdminEnforced: true
-    requiredApprovingReviewCount: 1
-    requiresCodeOwnerReviews: true
-    requiresStrictStatusChecks: false
-    requiredStatusCheckContexts:
-      - dependencies (8)
```
Comment (ORabbit) [inline .github/sync-repo-settings.yaml:24]: I'm sure there is a reason, but don't we want to check all of these?

Comment (anguillanneuf) [reply .github/sync-repo-settings.yaml:24]: These are for the java7 branch, which we don't have.

Comment (anguillanneuf) [reply .github/sync-repo-settings.yaml:24]: Something like this: https://github.com/googleapis/java-pubsub/tree/java7
Some libraries need to maintain a long-term java7 support.

Comment (ORabbit) [reply .github/sync-repo-settings.yaml:24]: Oh yes I see now. I missed the pattern main above.

**Review by ORabbit — APPROVED** [review]

**Pull request merged**